src/App.js

```diff
@@ -29,22 +29,82 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PlAYER_1);
+    // Wave 2
+    //  will need to create a method to change the square
+    //   When it is clicked on.
+    //   Then pass it into the squares as a callback
+  const squareClick = (id) => {
+    let madeMove = false;
+    const updateState = squares.map(row => row.map(pos => {
+      if (pos.id !== id) {return pos;}
+      madeMove =true;
+      return {...pos, value: currentPlayer};
+    }));
+    if (madeMove){
+    setSquares(newState);
+    let newPlayer;
+    if (currentPlayer == PLAYER_1) {
+      newPlayer = PlAYER_2;
+    }
+    setCurrentPlayer();
+    }  
+  };
+
+    // to map every id to the row and column on the array
+    // the row for any number is going to be this formula
+    // and the column is also this formula
+    const row = parseInt(id / 3);
+    const col = id - 3 * parseInt(id / 3);
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+    
+    // will pass whoevers turn it is here
+
+    squares[row][col] = 'x';
+
+    // change the state
+    setSquares(squares);
```

Comment:
When you push your updates we went over this should be fixed. We remove this code because you were just testing some things out.